nautobot_floor_plan/forms.py

```diff
@@ -102,66 +85,94 @@ class Meta:
         if not self.instance.created:
             self.initial["x_axis_labels"] = get_app_settings_or_config("nautobot_floor_plan", "default_x_axis_labels")
             self.initial["y_axis_labels"] = get_app_settings_or_config("nautobot_floor_plan", "default_y_axis_labels")
-            self.x_letters = self.initial["x_axis_labels"] == choices.AxisLabelsChoices.LETTERS
-            self.y_letters = self.initial["y_axis_labels"] == choices.AxisLabelsChoices.LETTERS
-            self.initial["x_origin_seed"] = "A" if self.x_letters else "1"
-            self.initial["y_origin_seed"] = "A" if self.y_letters else "1"
+            self.axis_letters["x"] = self.initial["x_axis_labels"] == choices.AxisLabelsChoices.LETTERS
+            self.axis_letters["y"] = self.initial["y_axis_labels"] == choices.AxisLabelsChoices.LETTERS
+            self.initial["x_origin_seed"] = "A" if self.axis_letters["x"] else "1"
+            self.initial["y_origin_seed"] = "A" if self.axis_letters["y"] else "1"
         else:
-            self.x_letters = self.instance.x_axis_labels == choices.AxisLabelsChoices.LETTERS
-            self.y_letters = self.instance.y_axis_labels == choices.AxisLabelsChoices.LETTERS
+            self.axis_letters["x"] = self.instance.x_axis_labels == choices.AxisLabelsChoices.LETTERS
+            self.axis_letters["y"] = self.instance.y_axis_labels == choices.AxisLabelsChoices.LETTERS
 
-        if self.x_letters and str(self.initial["y_origin_seed"]).isdigit():
+        if self.axis_letters["x"] and str(self.initial["x_origin_seed"]).isdigit():
             self.initial["x_origin_seed"] = general.grid_number_to_letter(self.instance.x_origin_seed)
-        if self.y_letters and str(self.initial["y_origin_seed"]).isdigit():
+        if self.axis_letters["y"] and str(self.initial["y_origin_seed"]).isdigit():
             self.initial["y_origin_seed"] = general.grid_number_to_letter(self.instance.y_origin_seed)
 
-        # Load existing custom ranges
+        # Initialize formsets for X and Y axis ranges
         if self.instance.pk:
-            x_ranges = list(
-                self.instance.custom_labels.filter(axis="X")
-                .values("start_label", "end_label", "step", "increment_letter", "label_type", "order")
-                .order_by("order")
-            )
-            y_ranges = list(
-                self.instance.custom_labels.filter(axis="Y")
-                .values("start_label", "end_label", "step", "increment_letter", "label_type", "order")
-                .order_by("order")
-            )
+            x_initial = [
+                {
+                    "start": range.start_label,
+                    "end": range.end_label,
+                    "step": range.step,
+                    "label_type": range.label_type,
+                    "increment_letter": range.increment_letter,
+                }
+                for range in self.instance.custom_labels.filter(axis="X").order_by("order")
```

Comment:
Given `range` is a built-in function, maybe use a variable like `label_range` instead?